api/app/src/models/medical/facility.ts

```diff
@@ -0,0 +1,51 @@
+import { DataTypes, Sequelize } from "sequelize";
+import { OIDNode, OID_ID_START } from "../../shared/oid";
+import { BaseModel, defaultModelOptions, ModelSetup } from "../_default";
+import { Organization } from "./organization";
+
+export class Facility extends BaseModel<Facility> {
+  static NAME = "facility";
+  declare id: number;
+  declare organizationId: number;
+  declare systemRootOid: string;
+
+  static setup: ModelSetup = (sequelize: Sequelize) => {
+    Facility.init(
+      {
+        ...BaseModel.baseAttributes(),
+        id: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+        },
+        organizationId: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+        },
+        systemRootOid: {
+          type: DataTypes.STRING,
+          primaryKey: true,
+        },
```

Comment:
Can't we have a primary ID of our own, and have a constraint with these columns to prevent duplication?

Composite DB keys make it more complex to work within the application and build queries. Everywhere we'd need to refer to 3 columns, even when its something internal to the application - I understand that user facing stuff requires them to provide these 3 columns, but stuff like building a select input is much easier with one column.

Also, `systemRootOid` should be always the same for a given install, right? So this would be using index resources without adding value to it. Could leave this as an optional customization to this project for those who want to make use of this repo on a multi-tenant setup on root OID level.

Could even use a sequential integer ID (since we don't plan to expose this ID anywhere like URLs and such). That's what HAPI uses and its good on indexes.